### ISSUE TITLE
Cookies are not automatically carried in redirection

### ISSUE THREAD

**Issue by au-top** [axios/axios#3862]
#### Describe the bug

I was on a sunny afternoon and I used axios to send a POST request to a website, and that website would redirect and use cookies to save the state in the middle... The tragedy happened here, and axios was automatically processing heavy After the orientation (302), the cookie was not saved .

Why not provide the option to open cookies in axios

I am using nodejs

Why are cookies prohibited in nodejs

#### To Reproduce
Code snippet to reproduce, ideally that will work by pasting into something like https://npm.runkit.com/axios, a hosted solution, or a repository that illustrates the issue. **If your problem is not reproducible, please file under Support or Usage Question**

```js
// Example code here
```

#### Environment
 - Axios Version [e.g. 0.18.0]
 - Adapter [e.g. XHR/HTTP]
 - Browser [e.g. Chrome, Safari]
 - Browser Version [e.g. 22]
 - Node.js Version [e.g. 13.0.1]
 - OS: [e.g. iOS 12.1.0, OSX 10.13.4]
 - Additional Library Versions [e.g. React 16.7, React Native 0.58.0]

#### Additional context/Screenshots
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by jgriffin1** [comment]
I am having this same issue as well. 
I am working on a node project that will be able to test other websites for being up and down. At the moment, I am attempting to do a post request to an external site with some credentials in the params, and this is failing because cookies don't seem to be readable after redirects.
The same post request works perfectly in Postman and Insomnia. 
After performing the request in postman, I can actually go to the console in postman, and then copy and paste the headers/url from the last redirect into my node js axios code and it works as long as I remember to include the cookie.

**Comment by ricreis394** [comment]
Have the same issue
I can POST correctly in Postman, but the same command in axios doesn't redirect well, resulting in unauthenticated response

**Comment by raymondKelly** [comment]
I am hitting the same issue. I found https://github.com/axios/axios/issues/953 which looks to be calling out the same issue. The response was basically that it is by design and that if you want to handle passing on headers/cookies in a redirect you should use the max redirects option and do something like 

`Axios.request({
    method: "get",
    url: "https://httpbin.org/cookies/set?a=1",
    maxRedirects: 0,
    validateStatus: function(status) {
        return status >= 200 && status < 303;
    },
}).then(res => {
    console.log(res.headers["set-cookie"]);
});`

**Comment by marioscialpi** [comment]
@raymondKelly works only in node.js

**Comment by raymondKelly** [comment]
> @raymondKelly
> 
> maxRedirects: 0
> 
> works only in node.js

Correct, perhaps I'm wrong on this but isn't this the NodeJS repository? If you are referring to it not work in the browser that is likely because the browser follows redirects

**Comment by dakriy** [comment]
Since this does not have any useful information for solving this issue and I had a hard time figuring this out here is what I did. 

Axios does not have a cookie jar built in as the last comment said, browsers do that. So we just have to handle cookies outside of Axios.

So I installed a custom httpAgent into axios that handles sending and setting cookies and it worked like a charm. Here is is what I used and it even has examples for integration into Axios.

https://www.npmjs.com/package/http-cookie-agent

**Comment by daniel-cues** [comment]
@dakriy would this work if I just want the browser to use already existing cookies in a third party site redirection? I'm unable to do that right now with axis
